File: backend/Entities/Inventory.cs
using System.Collections.Generic;
using Yantrik.Common;

namespace Yantrik.Entities
{
    public class Vendor : BaseEntity
    {
        public string CompanyName { get; set; } = string.Empty;
        public string? ContactPerson { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }
        public string? Address { get; set; }

        public ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
    }

    public class Category : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        
        public ICollection<Part> Parts { get; set; } = new List<Part>();
    }

    public class Part : BaseEntity
    {
        public string SKU { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal CostPrice { get; set; }
        public int StockQuantity { get; set; }
        public int MinThreshold { get; set; } = 10;

        public Guid CategoryId { get; set; }
        public Category Category { get; set; } = null!;
        
        public ICollection<InvoiceItem> InvoiceItems { get; set; } = new List<InvoiceItem>();
    }
}

File: frontend/src/lib/validations/inventory.ts
import { z } from 'zod';

export const createCategorySchema = z.object({
  name: z.string().min(2, 'Category name is required').max(50),
  description: z.string().optional(),
});

export type CreateCategoryFormValues = z.infer<typeof createCategorySchema>;

export const createPartSchema = z.object({
  name: z.string().min(2, 'Part name is required').max(100),
  sku: z.string().min(1, 'SKU is required').max(50),
  unitPrice: z.coerce.number().min(0, 'Must be 0 or more'),
  costPrice: z.coerce.number().min(0, 'Must be 0 or more'),
  stockQuantity: z.coerce.number().int().min(0, 'Must be 0 or more'),
  minThreshold: z.coerce.number().int().min(0).default(10),
  categoryId: z.string().min(1, 'Category is required'),
  description: z.string().optional(),
});

export type CreatePartFormValues = z.infer<typeof createPartSchema>;
